rllib/models/experimental/torch/primitives.py

```diff
@@ -0,0 +1,96 @@
+from typing import List, Optional
+from typing import Tuple
+
+from ray.rllib.models.experimental.base import Model
+from ray.rllib.models.specs.checker import (
+    input_is_decorated,
+    is_output_decorated,
+)
+from ray.rllib.models.temp_spec_classes import TensorDict
+from ray.rllib.utils.framework import try_import_torch
+from ray.rllib.utils.typing import TensorType
+from ray.rllib.models.experimental.base import ModelConfig
+
+torch, nn = try_import_torch()
+
+
+def _forward_not_decorated(input_or_output):
+    return (
+        f"forward not decorated with {input_or_output} specification. Decorate "
+        f"with @check_{input_or_output}_specs() to define a specification. See "
+        f"BaseModel for examples."
+    )
+
+
+class TorchModel(nn.Module, Model):
+    """Base class for torch models.
+
+    This class is used to define the general interface for torch models and checks
+    whether inputs and outputs are checked with `check_input_specs()` and
+    `check_output_specs()` respectively.
+    """
+
+    def __init__(self, config: ModelConfig):
+        nn.Module.__init__(self)
+        Model.__init__(self, config)
+        assert input_is_decorated(self.forward), _forward_not_decorated("input")
+        assert is_output_decorated(self.forward), _forward_not_decorated("output")
+
+    def forward(self, input_dict: TensorDict) -> Tuple[TensorDict, List[TensorType]]:
+        """Returns the output of this model for the given input.
+
+        Args:
+            input_dict: The input tensors.
+
+        Returns:
+            Tuple[TensorDict, List[TensorType]]: The output tensors.
+        """
+        raise NotImplementedError
+
+
+class FCNet(nn.Module):
+    """A simple fully connected network.
+
+    Attributes:
+        input_dim: The input dimension of the network. It cannot be None.
+        output_dim: The output dimension of the network. if None, the last layer would
+            be the last hidden layer.
+        hidden_layers: The sizes of the hidden layers.
+        activation: The activation function to use after each layer.
```

Comment:
Add explanation why we no longer support `activation` in combination with `last_layer_activation`. Now all layers, no matter what, have the same activation. Is this b/c we use separate pi-heads vs encoders (both could be FCNets in themselves) in the different algorithms anyways?

Reply:
We had a post_fcnet_activation before in vision net. Also the output_activation for attention. FullyConnectedNetwork didn't have an activation function on the final layer. I can't find something called exactly _last_layer_activation_.
But generally FCNet can also be an encoder. The final layers without activations will be appended automatically by whatever Models wraps this FCNet, independently of how the FCNet looks.

Reply:
We should support this and I've changed it!